Flank/flank
# Flank needs to respect the timeout value as that's a cap for billing purposes.

Fixes #836 


#### Flank billing calculation rules:

1. test execution time is rounded up to next minute and used in calculations

| Execution time  | Rounded value |
| ------------- | ------------- |
| `3m 22s` | `4m` |
| `15m 01s`  | `16m`  |
| `5m 00s` | `5m` |
2. if test execution time is smaller then test timeout (default 15m) round up execution time and use it in billing calculations
3. if test execution time is bigger then test timeout use timeout value in billing calculations
4. if test execution state is `ERROR` flank should skip billing calculation and assume `$0.00`

Examples:

1. execution time is ` 13m 24s` and timeout is default (`15m`), execution time is smaller then timeout therefor is rounded up to `14m` and used in billing calculations
2. execution time is `20m 13s` and timeout is set to `18m`, execution time is bigger then timeout therefor timeout value (`18m`) is used in billing calculations

## Test Plan
> How do we know the code works?

Flank needs to respect the timeout value as that's a cap for billing purposes, and round time to nearest minutes. 

1. Run flank with yml

```

gcloud:
  app: ./src/test/kotlin/ftl/fixtures/tmp/apk/app-debug.apk
  robo-script: ./src/test/kotlin/ftl/fixtures/test_app_cases/MainActivity_robo_script.json
  environment-variables:
    coverage: true
    coverageFilePath: /sdcard/
    clearPackageData: true

```
flank should return cost report with 2 min

```

CostReport
  Virtual devices
    $0.03 for 2m

```

2. Add in gcloud section timeout for 60s

```

gcloud:
  app: ./src/test/kotlin/ftl/fixtures/tmp/apk/app-debug.apk
  robo-script: ./src/test/kotlin/ftl/fixtures/test_app_cases/MainActivity_robo_script.json
  environment-variables:
    coverage: true
    coverageFilePath: /sdcard/
    clearPackageData: true
  timeout: 60s

```

flank should return cost report with 1 min
```

CostReport
  Virtual devices
    $0.02 for 1m

```

3. To check timeout is respected only when is lower than execution time set timeout to 5m and run

```

gcloud:
  app: ./src/test/kotlin/ftl/fixtures/tmp/apk/app-debug.apk
  robo-script: ./src/test/kotlin/ftl/fixtures/test_app_cases/MainActivity_robo_script.json
  environment-variables:
    coverage: true
    coverageFilePath: /sdcard/
    clearPackageData: true
  timeout: 5m

```

flank should return cost report with 2 min

```

CostReport
  Virtual devices
    $0.03 for 2m

```


## Checklist

- [X] release_notes.md updated
